src/app/space/create/deployments/apps/deployment-card.component.ts

```diff
@@ -129,23 +131,35 @@ export class DeploymentCardComponent implements OnDestroy, OnInit {
     }
   }
 
-  delete(): void {
+  openModal(): void {
+    this.modal.openModal(this);
+  }
+
+  delete(): boolean {
+    let succeeded = false;
     this.subscriptions.push(
-      this.deploymentsService.deleteDeployment(this.spaceId, this.applicationId, this.environment.name)
-        .subscribe(
+      this.deploymentsService.deleteDeployment(
+        this.spaceId,
+        this.environment.name,
+        this.applicationId
+        ).subscribe(
           (success: string) => {
             this.notifications.message({
               type: NotificationType.SUCCESS,
               message: success
             });
+            succeeded = true;
           },
           (error: any) => {
+            console.log(error);
             this.notifications.message({
               type: NotificationType.WARNING,
               message: error
             });
           }
         )
     );
+
+    return succeeded;
```

Comment:
Does this really work? You're returning synchronously and setting the flag within an async handler.